src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -283,6 +283,16 @@ def _check_config_and_set_final_flags(
                 f" Available names are: {', '.join(self._accelerator_types)}."
             )
 
+        # mps accelerator incompatible with ddp-family
```

Comment:
@awaelchli does Lite need this change too?

Reply:
Yes exactly. @shenoynikhil The equivalent change needs to be added to src/lightning_lite/connector.py
Feel free to add it this PR, or you can also leave it for a separate PR.

Reply:
I'll add it in this PR once the conversations are resolved.

Reply:
The tests seem to pass. I'm ok merging this as is and adding the corresponding changes for lightning_fabric/connector.py in a separate PR.

Reply:
Yes, I can create the other PR based once this merged.